### ISSUE TITLE
Launching EC2 Instance Fails v1.6.3

### ISSUE THREAD

**Issue by shakatechdayne** [ansible/ansible#7878]
Hi,

I am trying to use the ansible to launch an image in my AWS VPC, however whenever I try to use the ec2 module I get the following error:

```
failed: [localhost] => {"failed": true, "parsed": false}
invalid output was: Traceback (most recent call last):
  File "/home/ubuntu/.ansible/tmp/ansible-tmp-1403192659.01-129523988649411/ec2", line 2576, in <module>
    main()
  File "/home/ubuntu/.ansible/tmp/ansible-tmp-1403192659.01-129523988649411/ec2", line 1160, in main
    (instance_dict_array, new_instance_ids, changed) = create_instances(module, ec2)
  File "/home/ubuntu/.ansible/tmp/ansible-tmp-1403192659.01-129523988649411/ec2", line 956, in create_instances
    d = get_instance_info(inst)
  File "/home/ubuntu/.ansible/tmp/ansible-tmp-1403192659.01-129523988649411/ec2", line 574, in get_instance_info
    'ebs_optimized': inst.ebs_optimized}
AttributeError: 'Instance' object has no attribute 'ebs_optimized'
```

Here is the simple playbook I tried to use:

``` yaml

---
- name: Create a test server in AWS
  hosts: localhost
  gather_facts: false
  vars:
    keypair: SRVKey
    security_group: SRVBuild
    instance_type: t1.micro
    image: ami-XXXXXXX
    count: 1
    region: eu-west-1
    zone: eu-west-1a
    vpc_net: subnet-XXXXXXXX
  vars_prompt:
    privateip: "Please enter private IP"
    instance_name: "Please enter the instance name"
  tasks:
  - name: Launch instance
    local_action: 
      module: ec2 
      instance_tags: 
         Name: "{{ instance_name }}"
      key_name: "{{ keypair }}"
      group: "{{ security_group }}"
      instance_type: "{{ instance_type }}"
      image: "{{ image }}"
      vpc_subnet_id: "{{ vpc_net }}"
      private_ip: "{{ privateip }}"
      region: "{{ region }}"
      zone: "{{ zone  }}"
      count: "{{ count }}"
      wait: yes
```

This breaks on v1.6.3 however when I run this playbook on a v1.5.4 ansible install it works.

Thanks,

Dayne

**Comment by mpdehaan** [comment]
What version of boto do you have installed?  It sounds like you may need to upgrade.

That being said, we should also catch this and provide a proper error message.

**Comment by jimi-c** [comment]
I have already merged a patch in regarding this (1c3a654912f5058b527458eef5b94edf40ad43e4), but yes it is a boto version issue. Updating boto or using the devel branch will get you around this error. Thanks!
